### DISCUSSION TITLE
What's the behavior on merge conflicts?

### DISCUSSION THREAD

**Discussion by bakhtiyarneyman** [abraunegg/onedrive#1832] [Q&A] [answered]
I am worried about data loss in the event that a single file is modified from two different machines. What's the behavior in such case?

**Comment by abraunegg** [comment, marked as answer]
@bakhtiyarneyman 
> I am worried about data loss 

The number one thing here then is to ensure you build your system correctly:
* Ensure you are running the *latest* client - [![Version](https://img.shields.io/github/v/release/abraunegg/onedrive)](https://github.com/abraunegg/onedrive/releases) - or you are building the client from master. Do *not* rely on any older version. If you are building from source, the onus is on you to keep your client up-to-date. If using packages, use a distribution that keeps packages current, avoid those that do not.
* Use ZFS - mirrored pairs, raidz1,2,3 .. dont use anything else to store your data on, educate yourself as to why. Do not use anything else for your file system  for data you care about.
* Configure your 'sync_dir' to use a separate ZFS volume, configure tools and ZFS options to snapshot your data
* Configure the 'onedrive' application to handle situations where your 'drives' or ZFS volume are unavailable - look at all the options regarding '.nosync' so that the application stops syncing if / when that flag is exposed. This can be used in the exact situation you are wanting to avoid - that is, file system issues poisoning your OneDrive environment - see: https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#configuring-the-client-for-use-when-sync_dir-is-a-mounted-directory for details

> in the event that a single file is modified from two different machines. What's the behavior in such case?

If the newer file resides on OneDrive, and is *newer* by modified time or size + different based on calculated hash, if there is a file (same name. same path) locally, the local file will be renamed, and the file from OneDrive will be downloaded in its place preserving your local data.
